### PR TITLE
Artifact names and artifact youngest in the lineage

### PR DESCRIPTION
**Pull request by josenavas**
Adds the name property to the artifacts.

This has been build on top of #1587 so review/merge that one first.

### CODE REVIEW

**Comment by josenavas** [comment]
Merge conflict solved and now it also fixes #1573 (I've updated the title to reflect this change).

**Comment by josenavas** [comment]
After a discussion with @antgonza I'm going to also fix #1573 here - so hold on review.

**Comment by josenavas** [comment]
Talking again with @antgonza, including #1573 in this PR will make this PR to grow too much, so we decided to finally include it in another PR.

File: qiita_db/artifact.py [hunk 1]
```diff
@@ -255,8 +257,8 @@ def create(cls, filepaths, artifact_type, prep_template=None,
 
                 # Create the artifact
                 sql = """INSERT INTO qiita.artifact
-                            (generated_timestamp, command_id, data_type_id,
-                             command_parameters, visibility_id,
+                            (generated_timestamp, command_id,
```
Comment (ElDeveloper) [inline qiita_db/artifact.py]: Why was this changed?

File: qiita_db/artifact.py [hunk 2]
```diff
@@ -405,6 +410,46 @@ def delete(cls, artifact_id):
...
+            The new artifact's name
+
+        Raises
+        ------
+        ValueError
+            If `value` contains more than 35 chars
+        """
+        if len(value) > 35:
+            raise ValueError("The name of an artifact cannot exceed 35 chars. "
+                             "Current lengths: %d" % len(value))
```
Comment (ElDeveloper) [inline qiita_db/artifact.py]: lengths -> length

**Comment by ElDeveloper** [comment]
:+1:

File: qiita_db/artifact.py (hunk 1 above)
Comment (josenavas) [inline qiita_db/artifact.py]: For no good reason - reverting.

File: qiita_db/artifact.py (hunk 2 above)
Comment (josenavas) [inline qiita_db/artifact.py]: Done

**Comment by josenavas** [comment]
Thanks @ElDeveloper! comments addressed

File: qiita_db/artifact.py (hunk 1 above)
Comment (ElDeveloper) [inline qiita_db/artifact.py]: Thanks!

**Comment by antgonza** [comment]
Merging as the previous commit tests passed and the latest changes will not change that ...
